compiler/front/cli_reporter.nim

```diff
@@ -477,7 +477,8 @@ proc presentFailedCandidates(
         maybeWrongSpace = true
 
     for diag in err.diagnostics:
-      candidates.add reportFull(conf, diag) & "\n" # XXX: Tests fail because this is "Error: ..." and not "<Typename>: ..."
+      # Replace the "Error: " in the error messages with the name of the concept that the diagnostics concern
+      candidates.add reportFull(conf, diag).replace("Error:", err.diagnosticsTarget.name.s & ':') & '\n'
```

Comment:
Use `reportBody` and add missing parts of the message instead (via `conf.toStr(r.location.get())`).